pkg/proxy/iptables/proxier.go

```diff
@@ -266,7 +266,15 @@ func NewProxier(ipFamily v1.IPFamily,
 	masqueradeMark := fmt.Sprintf("%#08x", masqueradeValue)
 	klog.V(2).InfoS("Using iptables mark for masquerade", "ipFamily", ipt.Protocol(), "mark", masqueradeMark)
 
-	serviceHealthServer := healthcheck.NewServiceHealthServer(hostname, recorder, nodePortAddresses, healthzServer)
+	// get healthManager for the given IP Family
+	var healthManager healthcheck.ProxierHealthManager
+	if ipFamily == v1.IPv4Protocol {
+		healthManager = healthzServer.GetIPv4ProxierHealthManager()
+	} else {
+		healthManager = healthzServer.GetIPv6ProxierHealthManager()
+	}
```

Comment:
It would make more sense for the caller to pass the correct manager to `NewProxier()` rather than passing the whole `ProxyHealthServer` which it doesn't need.

Reply:
**NewProxier**() and **NewDualStackProxier**() will have different signatures then, for dual-stack then I would need to pass two health managers.